packages/node/src/login/oidc/refresh/TokenRefresher.spec.ts

```diff
@@ -24,44 +24,279 @@ import {
+      mocks.clientRegistrar ?? defaultMocks.clientRegistrar
     );
   }
 
-  it("Refreshes the token properly", async () => {
-    const key = "global";
-    const refreshTokenValue = "refresh token value";
+  it("throws if no oidc issuer can be retrieved from storage", async () => {
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:mySession": {
+        codeVerifier: "some code verifier",
+        redirectUri: "https://my.app/redirect",
+        dpop: "true",
+      },
+    });
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Failed to retrieve OIDC context from storage associated with session [mySession]"
+    );
+  });
+
+  it("throws if the token type cannot be retrieved from storage", async () => {
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:mySession": {
+        issuer: "https://my.idp",
+      },
+    });
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Failed to retrieve OIDC context from storage associated with session [mySession]"
+    );
+  });
+
+  it("throws if a refresh token isn't provided", async () => {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Missing a refresh token to refresh the access token associated to the session [mySession]"
+    );
+  });
+
+  it("throws if a dpop token is expected, but no dpop key is provided", async () => {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(
+      refresher.refresh("mySession", "some refresh token")
+    ).rejects.toThrow(
+      "The key bound to the access token associated to the session [mySession] must be provided to refresh it."
+    );
+  });
+
+  it("Refreshes a dpop token properly", async () => {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    const refreshedTokens = await refresher.refresh(
+      "mySession",
+      "some refresh token",
+      mockJwk()
+    );
+
+    expect(refreshedTokens.access_token).toEqual(mockDpopTokens().access_token);
+  });
 
-    const storageMock = mockStorageUtility(
-      {
-        [`solidClientAuthenticationUser:${key}`]: {
-          refreshToken: refreshTokenValue,
-        },
+  it("Refreshes a bearer token properly", async () => {
```

Comment:
```suggestion
  it("refreshes a bearer token properly", async () => {
```

Reply:
Resolved in 8818797c031a7e51ec3c56e1bf2e195d54ff6dee